Sources/Document/Json/JSONCounter.swift

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2022 The Yorkie Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+public protocol YorkieCountable: BinaryInteger {
+    var littleEndian: Self { get }
+    func addingReportingOverflow(_ other: Self) -> (partialValue: Self, overflow: Bool)
+}
+
+extension Int32: YorkieCountable {}
+extension Int64: YorkieCountable {}
+
+/**
+ * `JSONCounter` is a custom data type that is used to counter.
+ */
+public class JSONCounter<T: YorkieCountable> {
+    private let _value: T
```

Comment:
I don't know why JSONCounter holds the value in CRDTCounter.

Reply:
First of all, It is the same behavior with JS SDK.

When a user tries to create JSONCounter, SDK doesn't know the ChangeContext. So It can't make the CRDTCounter in init().  
The ChangeContext and CRDTCounter are created below and assigned by initialize()

1. set() function in JSONObject (line 88)
2. toJSONElelment() function in ElementConverter (line 56)
3. insertAfterInternal() function in JSONArray (line 311)

Before initialize() was called. _value is used as the temporal value.

Any better Idea?

Reply:
I have understood. Thank you for your detailed description.